hub/api/dataset.py

```diff
@@ -51,8 +84,16 @@ def __getitem__(self, item: Union[slice, str, int]):
     def __setitem__(self, item: Union[slice, str], value):
```

Comment:
so right now setitem is simply working as append_sample right instead of actually setting an item at an index?
we should consider renaming this if this behaviour will persist.

Reply:
No, that's not the case. Dataset's setitem just gives you the sliced dataset or the tensor. Tensor's setitem is either "write a new np.ndarray" or throw (it will eventually support update, but not append). Append will be implemented separately.